app/models/piece.rb

```diff
@@ -63,4 +63,25 @@ def is_obstructed?(start_point, end_point)
 		end
 	end
 
+	#get piece's current (start) position
+	def self.current_pos
```

Comment:
these methods should not be class level methods they should be instance level methods.  So remove self from each of these...